adr-0323-vestigial-clamps.md

```
@@ -0,0 +1,53 @@
+# ADR-0323: Remove Vestigial .min(u32::MAX) Clamps from usize→u32 Casts in diffguard-analytics
+
+## Status
```

markdownlint-cli2 (0.22.0)

[warning] Headings should be surrounded by blank lines
Expected: 1; Actual: 0; Below

(MD022, blanks-around-headings)